### DISCUSSION TITLE
v2.0.0

### DISCUSSION THREAD

**Discussion by gianni-rosato** [gianni-rosato/svt-av1-psy#35] [Announcements]
*Hello, everyone! We've been hard at work improving SVT-AV1 with our additions to the encoder improving visual fidelity. Little by little, we are working on trying to bring many of them to mainline! For the time being, I want to note that major SVT-AV1-PSY releases & mainline releases are not the same, and the codebases differ due to our changes; the version numbers may be identical, but the versions themselves are not, which is disclosed within the encoder's version information.*
With that, we're excited to announce **SVT-AV1-PSY v2.0.0**! 🎉

### Variance boost
- Moved varboost delta-q adjusting code to happen before TPL, giving TPL the opportunity to work with more accurate superblock delta-q priors, and produce better final rdmult lambda values
  - Fixed rare cases of pulsing at high CRFs (>=40) and strengths (3-4)
  - 2% avg. bitrate reduction for comparable image quality
- Added an alternative boosting curve (`--enable-alt-curve`), with different variance/strength tradeoffs
- Refactored boost code so it internally works with native q-step ratios
- Removed legacy variance boosting method based on 64x64 values
  - Parameter `--new-variance-octile` -> `--variance-octile`

*Excitingly, a var-boost mainline merge has been marked with the highest priority issue label by the mainline development team, so we may see this in mainline SVT-AV1 soon! Congrats @juliobbv!* 🎉 

### Other
- Presets got faster, so in addition to Preset -2, we have an even slower Preset -3
- `--sharpness` now accepts negative values
- The SVT-AV1-PSY encoder now supports Dolby Vision encoding via Dolby Vision RPUs. To build with Dolby Vision support, [install libdovi](https://github.com/quietvoid/dovi_tool/blob/main/dolby_vision/README.md#building-the-library) & pass `--enable-libdovi` to `./build.sh` on macOS/Linux (or `enable-libdovi` to `./build.bat` on Windows).

### Major API updates

- Changed the API signaling the End Of Stream (EOS) with the last frame vs with an empty frame
- `OPT_LD_LATENCY2` making the change above is kept in the code to help devs with integration
- The support of this API change has been [merged](https://github.com/FFmpeg/FFmpeg/commit/69dd1ce610fcffec453a0663c613c9b13165fd9e) to ffmpeg with a 2.0 version check
- Removed the 3-pass VBR mode which changed the calling mechanism of multi-pass VBR
- Moved to a new versioning scheme where the project major version will be updated every time API/ABI is changed

### Encoder

- Improve the tradeoffs for the random access mode across presets:
- Speedup presets MR by ~100% and improved quality along with tradeoff improvements across the higher quality presets ([!2179](https://github.com/AOMediaCodec/SVT-AV1/-/merge_requests/2179),[#2158](https://github.com/AOMediaCodec/SVT-AV1/-/issues/2158))
- Improved the compression efficiency of presets M9-M13 by 1-4% ([!2179](https://github.com/AOMediaCodec/SVT-AV1/-/merge_requests/2179))
- Simplified VBR multi-pass to use 2 passes to allow integration with ffmpeg
- Continued adding ARM optimizations for functions with `c_only` equivalent
- Replaced the 3-pass VBR with a 2-pass VBR to ease the multi-pass integration with ffmpeg
- Memory savings of 20-35% for LP 8 mode in preset M6 and below and 1-5% in other modes / presets

### Cleanup, bug fixes & documentation
- Various cleanups and functional bug fixes
- Update the documentation to reflect the rate control

*Thanks for using SVT-AV1-PSY!* ♥️

**Full Changelog**: https://github.com/gianni-rosato/svt-av1-psy/commits/v2.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/gianni-rosato/svt-av1-psy/releases/tag/v2.0.0'>v2.0.0</a>.</em>

**Comment by Patman86** [comment]
Is there a trick how SVT-AV1-PSY can be compiled in Windows with libdovi? Where should the library be located? I would like to test Dolby Vision. Batch File doesn't work!
